frontend/lib/src/components/shared/Icon/Icon.tsx

```diff
@@ -48,6 +48,7 @@ interface IconProps {
   color?: ThemeColor
   margin?: string
   padding?: string
+  testid?: string
 }
 
 const Icon = ({
```

Comment:
Adding `data-testid` to `Icon` component in `SidebarNav.tsx` doesn't pass down currently, so configured `Icon` to take a testid that is passed through

Reply:
How about we just add `data-testid="stIcon"` to here and then the test code would change to search based on index. I don't think we should add testid to the props.

Reply:
My main concern with making this a generic `data-testid="stIcon"` is that I can't find a way to test/distinguish between the ExpandMore Icon and the ExpandLess Icon 🤔 
I just know an Icon renders, but I can't test if its the correct one (Icon renders an svg w/ path vs an image with a src attribute thats testable):
<img width="435" alt="Screenshot 2023-09-19 at 2 56 52 PM" src="https://github.com/streamlit/streamlit/assets/63436329/f18e4462-caec-44e4-863d-524db098c9a9">

I would argue passing a testid as prop isn't fundamentally different from placing it as a prop directly in the component tho.

Reply:
Hm yeah I guess passing a testid as prop isn't fundamentally different. It just looked like an odd pattern 🤔 but I guess it's actually fine(?).